smurfs/src/reducers/reducers.js

```diff
@@ -0,0 +1,51 @@
+import { FETCH_SMURFS_START, FETCH_SMURFS_SUCCESS, FETCH_SMURFS_FAILURE, ADD_SMURFS, REMOVE_SMURFS } from "../actions/actions";
+export const initialState = {
+    smurfs: [{
+        name: "",
+        age: "",
+        height: "",
+        id: ""
+    }],
+    isFetching: false,
+    error: ""
+};
+
+const reducer = (state = initialState, action) => {
+    switch (action.type) {
+        case FETCH_SMURFS_START:
+            return {
+                ...state,
+                isFetching: true
+            };
+        case FETCH_SMURFS_SUCCESS:
+            return {
+                ...state,
+                smurfs: action.payload,
+                isFetching: false,
+                error: ''
+            };
+        case FETCH_SMURFS_FAILURE:
+            return {
+                ...state,
+                isFetching: false,
+                error: action.payload
+            };
+        case ADD_SMURFS:
+            return {
+                ...state,
+                smurfs: [...state.smurfs, action.payload]
```

Comment:
so check closely what your action.payload is here. If you look at your addSmurf dispatch function in the actions folder you can see that it is the res.data of the POST request right? If you console log that res.data you will see that it is an array that contains all the smurfs, including the one you just added. That means that when you put the action.payload inside this array here you are putting an array inside of an array. Since action.payload is already an array there is something much simpler you can do. Look at FETCH_SMURF_SUCCESS above for reference. Almost got it!